quarto/Group_10_Analysis.qmd

```diff
@@ -0,0 +1,281 @@
+editor: visual
+---
+
+# Introduction {#sec-intro}
+
+The research question is to examine whether the rating will be greater than 7 or not given the properties of films, such as length or duration, budget, votes, genre, and year of release.
+
+```{r}
+#| echo: true
+#| warning: false
+
+library(tidyverse)
+library(gt)
+
+# Read CSV from data dir
+df <- read_csv("../data/dataset10.csv")
+
+# Display the first 5 rows
+df |> 
+  slice_head(n=5) |>
+  gt() |>
+  cols_label(
+    film_id = html("Film ID"),
+    year = html("Year"),
+    length = html("Length"),
+    budget = html("Budget"),
+    votes = html("Votes"),
+    genre = html("Genre"),
+    rating = html("Rating")
+  )
+```
+
+Description:
+
+-   Film ID: The unique identifier for the film
+-   Year: Year of release of the film in cinemas
+-   Length: Duration (in minutes)
+-   Budget: Budget for the films production (in \$1000000s)
+-   Votes: Number of positive votes received by viewers
+-   Genre: Genre of the film
+-   Rating: IMDB rating from 0-10
+
+```{r}
+#| echo: true
+#| warning: false
+
+library(skimr)
+library(knitr)
+library(corrplot)
+library(ggplot2)
+library(gridExtra)
+library(dplyr)
+library(stats)
+library(jtools)
+library(sjPlot)
+library(broom)
+library(huxtable)
+```
+
+# Data Wrangling
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Calculate the number of NA for each column
+
+# Checking NAs
+na_sum <- colSums(is.na(df))
+
+# Plotting number of NAs
+bp <- barplot(na_sum, 
+        main = "Missing Values Count", 
+        ylab = "Count",                
+        col = "skyblue",              
+        names.arg = colnames(df), 
+        ylim = c(0, max(na_sum) + 200),
+        las=1) 
+text(x = bp, y = na_sum + 2, 
+     labels = na_sum, pos = 3, 
+     col = "red")
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Number of unique values in the categorical variables
+
+# Convert film_id as character
+df[c("film_id")] <- lapply(df[c("film_id")], as.character)
+
+# Compute number of unique values
+categorical_columns <- sapply(df, 
+                              function(x) is.factor(x) || is.character(x))
+unique_values <- apply(df[, categorical_columns], 
+                       2, 
+                       function(x) length(unique(x)))
+
+# Convert the output to be DataFrame
+columns.freq <- table(unique_values)
+columns.freq <- as.data.frame(columns.freq)
+colnames(columns.freq) <- c("columns", "number")
+
+# Creating barplot
+bp <- barplot(unique_values,
+              main='the number of unique values',
+              col='skyblue',
+              names.arg=colnames(unique_values),
+              ylim = c(0, max(unique_values) + 200),
+              legend.text = colnames(unique_values),
+              las=1)
+text(x = bp,
+     y = unique_values+75,
+     labels = unique_values)
+```
+
+```{r}
+#| echo: true
+#| warning: false
+
+# Create new columns: rating_higher_than_7
+df <- df %>% 
+  mutate(rating_higher_than_7 = ifelse(rating <= 7, 0, 1))
```

Comment:
Set a new column named rating_higher_than_7. If rating is higher than 7 ,the value will be 1, or it will be 0.